### DISCUSSION TITLE
How to select candidate equations using performance of test data more efficiently?

### DISCUSSION THREAD

**Discussion by leelew** [MilesCranmer/PySR#622] [Q&A] [answered]
Hi,

Thanks for developing such a useful tool! 

I try to discover equations from observational data. I run PySR with different parameters settings (e.g., complexity, operators), and I want to select equations according to the performance of test data (e.g., RMSE < a & R > b). But I have to select the equations manually which is time-costing. Is there any method to select candidate equations more efficiently?

Best regards,
Lu

**Comment by MilesCranmer** [comment, marked as answer]
Thanks!

Would the following help?

```julia
import copy

equations = copy.deepcopy(model.equations_)

# this is a pandas dataframe, so we can add new columns:
equations["my_metric"] = [
    my_metric(
        model.predict(Xtest, index=i),
        ytest
    )
    for i in range(len(equations))
]


choice = equations["my_metric"].idxmin()
# ^ or idxmax() if maximizing

model.predict(X, index=index)
# ^ Predict with best (or can pass to .sympy/.latex/.jax/.pytorch)
```
